LIMS2DB/classes.py

```diff
@@ -770,10 +770,16 @@ def get_library_preps(self, sample):
                         self.obj['samples'][sample.name]['library_prep'][prepname]['prep_status'] = inp_artifact.qc_flag
                         self.obj['samples'][sample.name]['library_prep'][prepname]['library_validation'][agrlibval.luid]['well_location'] = inp_artifact.containerplacement.api_string
                         self.obj['samples'][sample.name]['library_prep'][prepname]['library_validation'][agrlibval.luid]['reagent_labels'] = [rg.name for rg in inp_artifact.reagentlabels]
-                        if 'By user' not in self.obj['details']['library_construction_method'] and inp_artifact.reagentlabels:
+                        if 'By user' not in self.obj['details']['library_construction_method'] and len(inp_artifact.reagentlabels)==1:
                             # if finlib, these are already computed
                             self.obj['samples'][sample.name]['library_prep'][prepname]['reagent_label'] = inp_artifact.reagentlabels[0].name
                             self.obj['samples'][sample.name]['library_prep'][prepname]['barcode'] = self.extract_barcode(inp_artifact.reagentlabels[0].name)
+                        elif 'By user' not in self.obj['details']['library_construction_method'] and len(inp_artifact.reagentlabels)>1:
+                            # For cases that samples are indexed and pooled prior to Library QC
+                            for iaa in inp_artifact.ancestors:
+                                if iaa.reagentlabels and len(iaa.samples)==1 and iaa.samples[0].name==sample.name:
```

Comment:
Do you need an else-case for this or is it fine to just ignore other cases?

Reply:
The else case is for finished libraries, and it has been handled in Line 712-723

Reply:
I meant the `if` on line 780. Was that the one you answered on as well, or was it the one on line `777`?

Reply:
The `else` cases of line 780 are irrelevant artefacts which can be ignored.

Reply:
Ok, that makes sense! I'm not really comfortable with this codebase yet...